### PR TITLE
feat: reworked UI and made UI available in Linux version

### PR DESCRIPTION
**Pull request by hanneshapke**
## Description

Screenshot 2026-06-21 at 1 32 27 PM

## Linked Issue

Closes #

## Type of Change

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor / code quality
- [ ] Documentation update
- [ ] CI / build change

## Testing Done

**Unit tests (Go):**
```
make test-go
```

**Unit tests (Python):**
```
make test
```

**Linting & type checks:**
```
make lint
make typecheck
```

**Manual testing (if applicable):**
- Describe the scenario tested (e.g. PII type detected, proxy request flow, Chrome extension behavior)
- Include before/after behavior if fixing a bug
- Note any edge cases verified (e.g. nested PII in JSON payloads, restoration accuracy)

## Checklist

- [ ] `make test-go` and `make test` pass locally
- [ ] `make lint` reports no new errors
- [ ] New PII detection changes are covered by unit tests in `tests/`
- [ ] Go proxy changes include test coverage in `src/` or `model/`
- [ ] Documentation updated if behavior or configuration changed
- [ ] No sensitive data or credentials included

### CODE REVIEW

**Review by Davidnet — COMMENTED** [review]
Review of the new dashboard/web-UI backend. Two findings were reproduced at runtime against a locally-built dev backend (notes inline). The frontend↔backend contract tracing was otherwise clean.

**Not anchored to a line:**

- **System proxy not restored on shutdown (observed while running this build).** Starting the dev backend set the macOS system auto-proxy to `http://localhost:9090/proxy.pac`; a graceful `SIGTERM` did *not* tear it down (`networksetup -getautoproxyurl Wi-Fi` still showed `Enabled: Yes` after exit), leaving the machine pointed at a dead PAC server until manually disabled. Worth confirming whether the `transparent.go` changes affect the shutdown/restore path or whether it's a pre-existing dev-mode hazard.

**Review by Davidnet — COMMENTED** [review]
Refreshed review. The earlier inline findings are now **fixed** and merged into this branch:

- masked_rate=0, range=all OOM, swallowed timestamp parse, per-poll efficiency → #565
- transparent `/api/dashboard` route + auth-invariant test, system-proxy restore on shutdown → #566
- first-launch admin view → #567
- `truncatePreview` UTF-8 → #568

I've deleted those resolved comments. The items below are **still open** — they're pre-existing design/policy calls in this feature, not regressions.

**Partially addressed:** the divergent auth models (main mux is default-deny via `basicAuthMiddleware`; the transparent listener is default-allow via `isTransparentAdminProtectedPath`). #566 added a test asserting every transparent admin route is also in the protected list, which prevents a *new* route from silently shipping unauthenticated — but the two models still aren't structurally unified. Folding both listeners behind one default-deny gate would remove the foot-gun entirely.

**Release note:** PR carries no changeset, so merging won't bump the version / changelog — to be added separately before the release PR.

File: src/backend/server/server.go [hunk 1]
```diff
@@ -1021,13 +1110,110 @@ func (s *Server) handleModelInfo(w http.ResponseWriter, r *http.Request) {
...
 		log.Fatalf("Failed to start server: %v", err)
 	}
 }
 
+// basicAuthPublicPrefixes and basicAuthPublicExact list the request paths that
+// stay reachable WITHOUT Basic Auth even when it is enabled: the LLM proxy routes
+// (they carry their own provider API keys), all /api/pii/* endpoints (used by the
+// Chrome extension / API clients), and the lightweight health/version probes used
+// by load balancers (including their deprecated bare aliases).
+var basicAuthPublicPrefixes = []string{"/v1/", "/v1beta/", "/api/pii/"}
```
Comment (Davidnet) [inline src/backend/server/server.go:1143]: **Open / policy decision:** with Basic Auth enabled, `/v1/`, `/v1beta/`, and **all** `/api/pii/*` endpoints stay reachable unauthenticated. On a network-exposed Linux server that means anyone on the network can drive the proxy (`/v1`) and submit arbitrary text to the detection model (`/api/pii/check`, `/confidence`, `/entities`) — only the UI/admin endpoints are gated. If that's the intended contract (proxy carries provider keys; `/api/pii` is for the Chrome extension), worth a comment saying so explicitly; otherwise these exemptions should be scoped (e.g. by network/loopback) rather than always-open.

File: src/backend/server/server.go [hunk 2]
```diff
@@ -232,11 +246,23 @@ func (s *Server) Start() error {
...
 	// Add admin endpoints (e.g. health check, logs, certs, etc.)
 	mux := http.NewServeMux()
+	mux.HandleFunc("/api/health", s.healthCheck)
+	mux.HandleFunc("/api/version", s.versionHandler)
+	mux.HandleFunc("/api/auth/status", s.authStatusHandler)
+	mux.HandleFunc("/api/logs", s.logsHandler)
+	mux.HandleFunc("/api/mappings", s.mappingsHandler)
+	mux.HandleFunc("/api/stats", s.statsHandler)
+
+	// Deprecated: bare admin paths, kept only for backwards compatibility with
```
Comment (Davidnet) [inline src/backend/server/server.go:256]: **Open / convention:** these bare aliases (incl. `/version`) are marked "Deprecated … TODO: remove once existing extensions have migrated." The repo `CLAUDE.md` states `/version` is part of the public contract ("`server/server.go` — health/version endpoints (`/version` is part of the public contract)"). Removing a public-contract endpoint would break external clients. Either keep `/version` indefinitely (drop the removal TODO) or update the contract doc to reflect the migration to `/api/version`.

File: src/backend/server/server.go [hunk 3]
```diff
@@ -1021,13 +1110,110 @@ func (s *Server) handleModelInfo(w http.ResponseWriter, r *http.Request) {
...
+func (s *Server) validBasicAuth(r *http.Request) bool {
+	user, pass, ok := r.BasicAuth()
+	if !ok {
+		return false
+	}
+	wantUser := sha256.Sum256([]byte(s.config.BasicAuth.Username))
+	wantPass := sha256.Sum256([]byte(s.config.BasicAuth.Password))
+	gotUser := sha256.Sum256([]byte(user))
+	gotPass := sha256.Sum256([]byte(pass))
+	return subtle.ConstantTimeCompare(gotUser[:], wantUser[:]) == 1 &&
```
Comment (Davidnet) [inline src/backend/server/server.go:1192]: **Open / minor:** the `&&` short-circuits the password compare when the username hash differs, so an attacker can distinguish "wrong username" (one compare) from "valid username, wrong password" (two compares) by timing — a username-enumeration oracle, despite the "constant time" intent. Both inputs are already sha256'd, so the leak is small and network jitter dominates; if you want it fully constant-time, compute both `ConstantTimeCompare`s into ints and `&` (not `&&`) them so both always run.

File: src/frontend/src/components/settings/PIISection.tsx [hunk 4]
```diff
@@ -0,0 +1,439 @@
...
+    setSelectedRegexIndex(null);
+    setCustomRegexes(next);
+    persistRegexes(next);
+  };
+
+  // Values are the minimum confidence threshold to mask. A LOWER threshold
+  // catches more (higher sensitivity), so "High" sensitivity maps to 0.1 and
+  // "Low" to 0.5. Medium (0.25) stays the default.
+  const SENSITIVITY = [
+    { label: "Low", value: 0.5 },
```
Comment (Davidnet) [inline src/frontend/src/components/settings/PIISection.tsx:239]: **Open / behavior-change note (not a bug):** these presets invert the old modal's mapping (old Low=0.1 / High=0.5 → new Low=0.5 / High=0.1). Since the persisted value is the raw threshold, existing configs keep their actual masking behavior — only the label changes — but a returning user's stored 0.1 now renders as "High" instead of "Low," and clicking the same-labeled button now sends the opposite aggressiveness. The new mapping is arguably more correct (lower threshold = higher sensitivity); just worth a release note so the relabel isn't surprising.

**Review by Davidnet — APPROVED** [review]
